core/conversion/converters/impl/expand.cpp

```diff
@@ -150,22 +120,51 @@ bool add_expand_dynamic(
     ConversionCtx* ctx,
     const torch::jit::Node* n,
     nvinfer1::ITensor* in,
-    nvinfer1::ITensor* expandedDimsTensor) {
-  auto input_shape_tensor = ctx->net->addShape(*in)->getOutput(0);
+    nvinfer1::ITensor* expandedDimsTensor,
+    nvinfer1::Dims expandedDims,
+    bool is_expand_layer) {
+  auto input_dims = in->getDimensions();
   auto input_rank = in->getDimensions().nbDims;
   auto output_rank = expandedDimsTensor->getDimensions().d[0];

```

Comment:
Sorry, the "output_rank" should be the number of output dimensions. It was my variable name that mislead you. expandedDimsTensor->getDimensions().d[0] is equal to expandedDims.nbDims. Eg: an expandedDimsTensor is [2, 1, 2], the dimensions of expandedDimsTensor is Dims{1, {3}}, and an expandedDims is Dims{3, {2, 1, 2}}.  Then expandedDimsTensor->getDimensions().d[0] is equal to 3 and equal to expandedDims.nbDims.  Does my explanation help you?

Reply:
Thank you for your explanation. I understand. Can we use .nbDims directly for easy understanding for new users ? Since we already do getDimensions() in the RegisterNodeConversionPatterns, we may not need to get a shape vector of the dimensions and access the first element again here right ?